### ISSUE TITLE
add some utility functions for hash partitioning

### ISSUE THREAD

**Issue by digoal** [postgrespro/pg_pathman#49]
HI,
   there is some wrapped function for range partition , like add/attach/detach/remove partition . 
   but there is no wrapped funcs for hash partition, if some body like to use pg_pathman for sharding(based on fdw), i think these funcs will help them.
   thanks, this an advise.

**Comment by zilder** [comment]
Hi @digoal,

Good point! I think we'll come up with something soon to support this.

**Comment by zilder** [comment]
Hi @digoal,

I've added a new function which replaces one of the hash partitions with another table, so that you for example could replace a partition with a foreign table.

```sql
replace_hash_partition(
	old_partition		REGCLASS,
	new_partition		REGCLASS)
```

This hasn't been merged into `master` branch yet, but if you would like to check it out, please clone the `hash_replace` branch, rebuild it and recreate an extension.

**Comment by zilder** [comment]
Simple example:

```
create table test(id serial);
select create_hash_partitions('test', 'id', 2);
create table test_external (like test including all);
select replace_hash_partition('test_0', 'test_external');
explain select * from test;
                              QUERY PLAN                               
-----------------------------------------------------------------------
 Append  (cost=0.00..71.00 rows=5100 width=4)
   ->  Seq Scan on test_external  (cost=0.00..35.50 rows=2550 width=4)
   ->  Seq Scan on test_1  (cost=0.00..35.50 rows=2550 width=4)
(3 rows)
```

**Comment by zilder** [comment]
This feature was merged into `master` and now available in `1.2` release.
